PostTaskSecurity.md

```diff
@@ -0,0 +1,179 @@
+# `scheduler.postTask` Security Considerations
+
+The main security concern is whether or not the API adds new surfaces to
+perform side-channel attacks, specifically for same-process cross-origin
+iframes. This might be a concern because `postTask`
+
+1. exposes priorities, which influence the order in which tasks run
+2. allows tasks to be posted with a delay
+
+The two main attacks we consider here are whether `postTask` leaks any new
+information through priorities, and  whether prioritized delayed tasks can be
+used as a new high-resolution timing source.
+
+## Inferring Information from Task Timing
+
+Script can attempt to learn something about other tasks in the system by
+measuring timing of the tasks it controls, specifically the delay [1]. In
+other words, an attacker can queue tasks and try to infer something about the
+system based on the time the tasks started to run. The relevant question here
+is, can a frame learn anything (interesting) about other frames running in the
+same process by exploiting the ordering guarantees of prioritized tasks?
+
+[1] Task duration might provide some information about the underlying system,
+specifically the OS, but is beyond the scope of this API since that is
+fundamental to all tasks, not the ones introduced here.
+
+### Recap of Prioritization
+
+The browser has freedom to choose between tasks in different task queues, which
+correspond to different task sources (e.g. network vs. user input). The
```

Comment:
Is it worth linking to the spec definition task sources here?

Reply:
Yeah good suggestion; I just added a link to the task source definition and to the event loop processing model where task queue selection is defined.